### PR TITLE
ref(trace) rename bad casing and make method types consistent 

### PR DESCRIPTION
**Pull request by JonasBa**
Minor cleanup of the banner component and moving the usage api hook to the api folder

### CODE REVIEW

File: static/app/views/performance/newTraceDetails/traceModels/traceTree.spec.tsx [hunk 1]
```diff
@@ -2323,7 +2323,7 @@ describe('TraceTree', () => {
       expect(root.children.length).toBe(1);
 
       assertAutogroupedNode(root.children[0]);
-      expect(root.children[0].has_errors).toBe(true);
```
Comment (JonasBa) [inline static/app/views/performance/newTraceDetails/traceModels/traceTree.spec.tsx:2326]: fixed up the bad casing on getters

File: static/app/views/performance/newTraceDetails/traceModels/traceTree.tsx [hunk 2]
```diff
@@ -124,7 +124,6 @@ import {cloneTraceTreeNode} from './traceTreeNodeHelpers';
  * - there is an annoying thing wrt span and transaction nodes where we either store data on _children or _spanChildren
  *   this is because we want to be able to store both transaction and span nodes in the same tree, but it makes for an
  *   annoying API. A better design would have been to create an invisible meta node that just points to the correct children
- * - instead of storing span children separately, we should have meta tree nodes that handle pointing to the correct children
```
Comment (JonasBa) [inline static/app/views/performance/newTraceDetails/traceModels/traceTree.tsx:127]: Duplicate comment of the one above

File: static/app/views/performance/newTraceDetails/traceModels/traceTree.tsx [hunk 3]
```diff
@@ -220,10 +219,6 @@ export type ViewManagerScrollToOptions = {
   organization: Organization;
 };
 
-function cacheKey(organization: Organization, project_slug: string, event_id: string) {
```
Comment (JonasBa) [inline static/app/views/performance/newTraceDetails/traceModels/traceTree.tsx:223]: unnecessary

File: static/app/views/performance/newTraceDetails/traceModels/traceTree.tsx [hunk 4]
```diff
@@ -270,7 +265,7 @@ function childParentSwap({
 
   // We need to remove the portion of the tree that was previously a child, else we will have a circular reference
   parent.parent = child;
-  child.children.push(parent.filter(parent, n => n !== child));
+  child.children.push(TraceTreeNode.Filter(parent, n => n !== child));
```
Comment (JonasBa) [inline static/app/views/performance/newTraceDetails/traceModels/traceTree.tsx:268]: Same static method as ForEach and friends

**Review by Abdkhan14 — APPROVED** [review]

**Pull request closed without merging**